crates/worker/src/partition/mod.rs

```diff
@@ -187,15 +197,22 @@ where
                         transaction.commit().await?;
                         state.handle_actions(action_collector.drain(..)).await?;
                     }
+                    histogram!(PP_APPLY_RECORD_DURATION, PARTITION_LABEL => partition_id_str).record(command_start.elapsed());
                 },
                 action_effect = action_effect_stream.next() => {
+                    histogram!(PP_WAIT_OR_IDLE_DURATION, PARTITION_LABEL => partition_id_str).record(iteration_start.elapsed());
+                    let effect_start = Instant::now();
                     counter!(PARTITION_ACTUATOR_HANDLED).increment(1);
                     let action_effect = action_effect.ok_or_else(|| anyhow::anyhow!("action effect stream is closed"))?;
                     state.handle_action_effect(action_effect).await?;
+                    histogram!(PP_APPLY_EFFECTS_DURATION).record(effect_start.elapsed());
```

Comment:
Technically, it is not the time it takes to apply the effects but to append them to Bifrost.

Reply:
I've removed most of those metrics. I'll change the strategy entirely in a future PR.